moviebuff.go

```diff
@@ -65,13 +68,14 @@ func New(config Config) Moviebuff {
 // Instead of the UUID, this can also be the URL of the movie as seen on moviebuff.com, like 12-years-a-slave
 // Details include release dates, certifications, cast, crew, trailers, posters, purchase links etc.
 // Here movies may include feature films, documentaries, short films etc.
-func (m *moviebuff) GetMovie(id string) (*Movie, error) {
-	r, err := prepareRequest(m.HostURL, m.StaticToken, "/resources/movies/"+id)
+func (m *moviebuff) GetMovie(ctx context.Context, id string) (*Movie, error) {
+	r, err := prepareRequest(ctx, m.HostURL, m.StaticToken, "/resources/movies/"+id)
 	if err != nil {
 		return nil, err
 	}
 
-	res, err := new(http.Client).Do(r)
+	client := apmhttp.WrapClient(http.DefaultClient)
```

Comment:
can you check we can wrap once and reuse the client

Reply:
Please let me know if below change looks good
```
type moviebuff struct {
	Config
	Client *http.Client
}

func New(config Config) Moviebuff {
	return &moviebuff{
		Config: config,
		Client: apmhttp.WrapClient(http.DefaultClient),
	}
}
```
While calling
```
func (m *moviebuff) GetMovie(ctx context.Context, id string) (*Movie, error) {
	res, err := m.Client.Do(r)
	if err != nil {
		return nil, err
	}
}
```